plugin/src/main/java/io/jenkins/plugins/remotingkafka/GlobalKafkaConfiguration.java

```diff
@@ -211,11 +288,132 @@ public FormValidation doTestBrokerConnection(@QueryParameter("brokerURL") final
+            @QueryParameter("kubernetesSkipTlsVerify") boolean skipTlsVerify,
+            @QueryParameter("kubernetesNamespace") String namespace
+    ) {
+        Jenkins.get().checkPermission(Jenkins.ADMINISTER);
+
+        try {
+            String serverUrl = new URIBuilder()
+                    .setHost(serverIp)
+                    .setPort(Integer.parseInt(serverPort))
+                    .toString();
+            KubernetesClient client = new KubernetesFactoryAdapter(serverUrl, namespace,
+                    Util.fixEmpty(serverCertificate), Util.fixEmpty(credentialsId), skipTlsVerify
+            ).createClient();
+            // Call Pod list API to ensure functionality
+            client.pods().list();
+            return FormValidation.ok("Success");
+        } catch (KubernetesClientException e) {
+            LOGGER.log(Level.FINE, "Error testing Kubernetes connection", e);
+            return FormValidation.error("Error: %s", e.getCause() == null
+                    ? e.getMessage()
+                    : String.format("%s: %s", e.getCause().getClass().getName(), e.getCause().getMessage()));
+        } catch (Exception e) {
+            LOGGER.log(Level.FINE, "Error testing Kubernetes connection", e);
+            return FormValidation.error("Error: %s", e.getMessage());
+        }
+    }
+
+    @RequirePOST
+    public FormValidation doStartKafkaOnKubernetes(
+            @QueryParameter("kubernetesIp") String serverIp,
+            @QueryParameter("kubernetesApiPort") String serverPort,
+            @QueryParameter("kubernetesCredentialsId") String credentialsId,
+            @QueryParameter("kubernetesCertificate") String serverCertificate,
+            @QueryParameter("kubernetesSkipTlsVerify") boolean skipTlsVerify,
+            @QueryParameter("kubernetesNamespace") String namespace
+    ) {
+        // TODO: Use K8s client to launch Zookeeper and Kafka pods
+        Jenkins.get().checkPermission(Jenkins.ADMINISTER);
+
+        try {
+            String serverUrl = new URIBuilder()
+                    .setHost(serverIp)
+                    .setPort(Integer.parseInt(serverPort))
+                    .toString();
+            KubernetesClient client = new KubernetesFactoryAdapter(serverUrl, namespace,
+                    Util.fixEmpty(serverCertificate), Util.fixEmpty(credentialsId), skipTlsVerify
+            ).createClient();
+            Class clazz = GlobalKafkaConfiguration.class;
+            client.load(clazz.getResourceAsStream("kubernetes/zookeeper.yaml")).createOrReplace();
+            LOGGER.info("Starting Zookeeper");
+            client.load(clazz.getResourceAsStream("kubernetes/kafka-service.yaml")).createOrReplace();
+            LOGGER.info("Starting Kafka Services");
+            Integer zookeeperPort = KubernetesUtils.getFirstNodePortByServiceName(client, "zookeeper-svc");
+            Integer kafkaPort = KubernetesUtils.getFirstNodePortByServiceName(client, "kafka-svc");
```

Comment:
I wonder if these port vars should be `int`...you should have null checks if you keep them `Integer`